pkg/util/crypto.go

```diff
@@ -53,9 +53,10 @@ func GenerateSelfSignedCert(host, certPath, keyPath string, alternateIPs []net.I
 		NotBefore: time.Now(),
 		NotAfter:  time.Now().Add(time.Hour * 24 * 365),
 
-		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
```

Comment:
No disagreement about self signed certs being useful. My point was that making all callers of this function that generate a self signed cert instead generate a self signed **certificate authority** cert doesn't seem right. If we want to do this just for the local cluster up case, then we'd need to plumb through a config switch so that we don't automatically inherit it for all callers of this function (for example, the kubelet currently generates a self signed cert for its https endpoint if it doesn't find one in the default location -- that cert shouldn't need to be a CA).